client/components/explorer/sidebar/sidebar-tabs-directive.css

```diff
@@ -23,4 +23,10 @@
 .pk-sidebar-tab-label {
   padding-left: 32px;
   padding-top: 2px;
+  /*
+   * Ensure there's always enough room to fit the largest-used label,
+   * currently this is "Data Results" which is longer than "Dashboard".
+   * This avoids unsightly relayout when showing/hiding labels.
+   */
+  min-width: 107px;
```

Comment:
The specific issue is that the left-hand bar changes size depending on which items are currently displayed, and this causes a content resize.

How to reproduce:

- open a dashboard. Left-hand bar contains "Dashboard" and "Container" items.
- click on a graph. Left-hand bar now gets additional items including "Data Results", and since this is wider than the previously-longest "Dashboard" label the left-hand bar resizes, shrinking the main content area and triggering resize.

I agree that the hardcoded size is ugly. I'll see if it's feasible to have the items always present with visibility: hidden so that they reserve space without being shown.